packages/identifier-resolution/src/types/IJwtService.d.ts

```diff
@@ -0,0 +1,226 @@
+export type Jwe = JweCompact | JweJsonFlattened | JweJsonGeneral;
+export type JweCompact = string;
+export interface JweJsonFlattened {
+    protected: string;
+    unprotected: JweHeader;
+    header: JweHeader | JweRecipientUnprotectedHeader;
+    encrypted_key?: string;
+    aad?: string;
+    iv: string;
+    ciphertext: string;
+    tag?: string;
+}
+export interface JweRecipient {
+    header?: JweRecipientUnprotectedHeader;
+    encrypted_key?: string;
+}
+export interface JweJsonGeneral {
+    protected: string;
+    unprotected?: JweHeader;
+    recipients: Array<JweRecipient>;
+    aad?: string;
+    iv: string;
+    ciphertext: string;
+    tag?: string;
+}
+export interface PreparedJwsObject {
+    jws: PreparedJws;
+    b64: {
+        payload: string;
+        protectedHeader: string;
+    };
+    identifier: ManagedIdentifierResult;
+}
+export interface BaseJwtHeader {
+    typ?: string;
+    alg?: string;
+    kid?: string;
+}
+export interface BaseJwtPayload {
+    iss?: string;
+    sub?: string;
+    aud?: string[] | string;
+    exp?: number;
+    nbf?: number;
+    iat?: number;
+    jti?: string;
+}
+export interface JwsHeader extends BaseJwtHeader {
+    kid?: string;
+    jwk?: JWK;
+    x5c?: string[];
+    [key: string]: unknown;
+}
+export interface JwsPayload extends BaseJwtPayload {
+    [key: string]: unknown;
+}
+export interface JwsHeaderOpts {
+    alg: JoseSignatureAlgorithm | JoseSignatureAlgorithmString;
+}
+export type JwsIdentifierMode = 'x5c' | 'kid' | 'jwk' | 'did' | 'auto';
+export type EncryptJweCompactJwtArgs = {
+    payload: JwsPayload;
+    protectedHeader?: JweProtectedHeader | undefined;
+    aad?: Uint8Array | undefined;
+    recipientKey: ExternalIdentifierResult & {
+        kid?: string;
+    };
+    alg?: JweAlg;
+    enc?: JweEnc;
+    apu?: string;
+    apv?: string;
+    expirationTime?: number | string | Date;
+    issuer?: string;
+    audience?: string | string[];
+};
+export type DecryptJweCompactJwtArgs = {
+    jwe: JweCompact;
+    idOpts: ManagedIdentifierOptsOrResult;
+};
+export type CreateJwsArgs = {
+    mode?: JwsIdentifierMode;
+    issuer: ManagedIdentifierOptsOrResult & {
+        noIssPayloadUpdate?: boolean;
+        noIdentifierInHeader?: boolean;
+    };
+    clientId?: string;
+    clientIdScheme?: ClientIdScheme | 'did' | string;
+    protectedHeader: JwsHeader;
+    payload: JwsPayload | Uint8Array | string;
+};
+export type CreateJweArgs = {
+    mode?: JwsIdentifierMode;
+    issuer: ManagedIdentifierOptsOrResult & {
+        noIssPayloadUpdate?: boolean;
+        noIdentifierInHeader?: boolean;
+    };
+    protectedHeader: JweProtectedHeader;
+    encryptedKey: string | EphemeralPublicKey;
+    iv: string;
+    ciphertext: string;
+    tag: string;
+};
+export type CreateJwsCompactArgs = CreateJwsArgs;
+export type CreateJwsFlattenedArgs = Exclude<CreateJwsJsonArgs, 'existingSignatures'>;
+export type VerifyJwsArgs = {
+    jws: Jws;
+    jwk?: JWK;
+    opts?: {
```

Comment:
maybe nicer to make a type for this?

Reply:
I do not control that, this is a ".d.ts" typings file copied from jwt-service because I can't add that package to any dependency list